File: exercises/pset6/credit/credit.py
from cs50 import get_string
while True:
    card = get_string("Number:")
    if(len(card) >= 13):
        if(len(card) <= 16):
            break
digit = 2
normal = 0
double = 0
for i in range(0,len(card)):
    if (digit % 2 == 0):
        normal += ord(card[len(card) - i - 1]) - 48
        digit+= 1
    elif ((ord(card[len(card) - i - 1]) - 48) * 2 > 9):
        double += (((ord(card[len(card) - i - 1]) - 48) * 2) % 10) + (((ord(card[len(card) - i - 1]) - 48) * 2) - (((ord(card[len(card) - i - 1]) - 48) * 2) % 10)) // 10
        digit += 1
    else:
        double += (ord(card[len(card) - i - 1]) - 48) * 2
        digit += 1
if ((double + normal) % 10 != 0):
    print("INVALID")
elif (((ord(card[0])-48) * 10 + (ord(card[1]) - 48)) == 37):
    print("AMEX")
elif (((ord(card[0])-48) * 10 + (ord(card[1]) - 48)) > 50 and ((ord(card[0])-48) * 10 + (ord(card[1]) - 48)) < 56):
    print("MASTERCARD")
else:
    print("VISA")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
int main(void)
{
    long i = 0;
    do
    {
        i = get_long("Números do cartão: ");
    }
    while (i < 0);
    if (i < 1000000000000) //less than 13 digits
    {
        printf("INVALID\n");
    }
    else
    {
        int a = 0;
        int b = 0;
        long j = i;
        do
        {
            //take digits one by one
            int g = i % 10;
            a = a + g;
            i = ((i - i % 10) / 10);
            g = i % 10;
            if (g >= 5)
            {
                b = b + ((g * 2) % 10) + ((2 * g - (g * 2) % 10) / 10);
            }
            else
            {
                b = b + 2 * g;
            }
            i = ((i - i % 10) / 10);
        }
        while (i > 0);
        int s = a + b;
        int k = ((s + 1) % 10);
        if (((s + 1) % 10) == 1)
        {
            do
            {
                j = (j - (j % 10)) / 10;
            }
            while (j > 99);
            //flag identifier
            if (51 <= j && j <= 55)
            {
                printf("MASTERCARD\n");
            }
            else if (j == 34 || j == 37)
            {
                printf("AMEX\n");
            }
            else if (j >= 40 && j <= 49)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
}